x-pack/metricbeat/module/sql/query/test_sql_mssql.py

```diff
@@ -0,0 +1,75 @@
+import os
+import sys
+import unittest
+import time
+from xpack_metricbeat import XPackTest, metricbeat
+
+
+class Test(XPackTest):
+    COMPOSE_SERVICES = ['mssql']
+
+    @unittest.skipUnless(metricbeat.INTEGRATION_TESTS, "integration test")
+    def test_query_without_fetch_from_all_dbs(self):
+        """
+        SQL MSSQL custom query with fetch_from_all_dbs=false
+        """
+        self.render_config_template(modules=[{
+            "name": "sql",
+            "metricsets": ["query"],
+            "hosts": ['sqlserver://{}:{}@{}'.format(self.get_username(), self.get_password(), self.compose_host())],
+            "period": "5s",
+            "additional_content": """
+  driver: mssql
+  fetch_from_all_dbs: false
+  sql_query: SELECT DB_NAME() AS 'database_name';
+  sql_response_format: table"""
+        }])
+
+        proc = self.start_beat()
+        self.wait_until(lambda: self.output_lines() > 0)
+        proc.check_kill_and_wait()
+        self.assert_no_logged_warnings()
+
+        output = self.read_output_json()
+        self.assertEqual(len(output), 1)
+
+        for evt in output:
+            self.assert_fields_are_documented(evt)
+            self.assertIn("sql", evt.keys(), evt)
+            self.assertIn("query", evt["sql"].keys(), evt)
+
+    @unittest.skipUnless(metricbeat.INTEGRATION_TESTS, "integration test")
+    def test_query_with_fetch_from_all_dbs(self):
+        """
+        SQL MSSQL custom query with fetch_from_all_dbs=true
+        """
+        self.render_config_template(modules=[{
+            "name": "sql",
+            "metricsets": ["query"],
+            "hosts": ['sqlserver://{}:{}@{}'.format(self.get_username(), self.get_password(), self.compose_host())],
+            "period": "5s",
+            "additional_content": """
+  driver: mssql
+  fetch_from_all_dbs: true
+  sql_query: SELECT DB_NAME() AS 'database_name';
+  sql_response_format: table"""
+        }])
+
+        proc = self.start_beat()
+        self.wait_until(lambda: self.output_lines() > 0)
+        proc.check_kill_and_wait()
+        self.assert_no_logged_warnings()
+
+        output = self.read_output_json()
+        self.assertEqual(len(output), 4)
+
+        for evt in output:
+            self.assert_fields_are_documented(evt)
+            self.assertIn("sql", evt.keys(), evt)
+            self.assertIn("query", evt["sql"].keys(), evt)
```

Comment:
Another approach you can take here is reducing redundacy by:
1) Making a helper method `run_sql_query_test(self, fetch_from_all_dbs)`
2) Call it from each test with `self.run_sql_query_test(fetch_from_all_dbs=boolean)`
3) Use f string formatting in `"additional_content"` for `fetch_from_all_dbs` converting the boolean: `str(fetch_from_all_dbs).lower()`
4) Use a one liner to validate `expected_output_count`:
>         expected_output_count = 4 if fetch_from_all_dbs else 1
>         self.assertEqual(len(output), expected_output_count)

Do you think this approach is worth it? @shmsr

Reply:
Sure. I'll make the change. Sorry, I'm not good with Python so that's why I couldn't write it effectively but thanks for your input. I'll push the code changes ASAP.

Reply:
Done!